realm/realm-library/src/androidTest/java/io/realm/GCTests.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2016 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+
+import android.support.test.runner.AndroidJUnit4;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import io.realm.entities.AllTypes;
+import io.realm.entities.Dog;
+import io.realm.rule.TestRealmConfigurationFactory;
+
+import static junit.framework.TestCase.assertNotNull;
+
+// This test is for the fact we don't have locks for native objects creation that when finalizer/phantom thread free the
```

Comment:
can we trigger/assert the collection of references (using the daemon thread) to ensure the assumption.
> the same Realm could have some native objects creation at the same time